staging/src/k8s.io/apiserver/pkg/server/options/authentication.go

```diff
@@ -80,6 +84,9 @@ func (s *RequestHeaderAuthenticationOptions) Validate() []error {
 	if len(s.UsernameHeaders) > 0 && !caseInsensitiveHas(s.UsernameHeaders, "X-Remote-User") {
 		klog.Warningf("--requestheader-username-headers is set without specifying the standard X-Remote-User header - API aggregation will not work")
 	}
+	if len(s.UIDHeaders) > 0 && !caseInsensitiveHas(s.UIDHeaders, "X-Remote-Uid") {
+		klog.Warningf("--requestheader-uid-headers is set without specifying the standard X-Remote-Uid header - API aggregation will not work")
```

Comment:
since this is a new header / flag, should we be more strict here and make this an error to start?

Reply:
Let's do it, that way we can always switch back to a warning in case somebody's able to find a plausible use case that doesn't use the standard header.